### PR TITLE
Implement OLAP edge filtering by image of GraphFilterInputFormat

### PR DESCRIPTION
**Pull request by YevIgn**
Fixes: #1083 

Signed-off-by: Evgeniy Ignatiev <yevgeniy.ignatyev@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by mbrukman** [comment]
CLA is addressed in https://github.com/JanusGraph/legal/pull/95

File: janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/util/GiraphRecordReader.java [hunk 1]
```diff
@@ -50,6 +59,12 @@ public GiraphRecordReader(final GiraphInputFormat.RefCountedCloseable<JanusGraph
     @Override
     public void initialize(final InputSplit inputSplit, final TaskAttemptContext taskAttemptContext) throws IOException, InterruptedException {
         reader.initialize(inputSplit, taskAttemptContext);
+
+        final Configuration conf = HadoopCompat.getConfiguration(taskAttemptContext);
```
Comment (jerryjch) [inline janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/util/GiraphRecordReader.java]: Use taskAttemptContext.getConfiguration(), not the HadoopCompat method which comes from Cassandra package.

Comment (YevIgn) [reply janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/util/GiraphRecordReader.java]: Thanks for your review! I totally overlooked HBaseSnapshotInputFormatIT.

**Comment by YevIgn** [comment]
Have no idea what to do with SolrIndexTest - it doesn't fail on my machine and I haven't changed any code that should affect it.

**Comment by YevIgn** [comment]
Opened issue https://github.com/JanusGraph/janusgraph/issues/1091 for the `SolrIndexTest#testTTL`.

**Comment by jerryjch** [comment]
I will commit it soon if there is no more comments.

**Pull request merged**